src/github.com/matrix-org/dendron/proxy/proxy.go

```diff
@@ -105,3 +110,21 @@ func (p *SynapseProxy) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	p.ProxyHTTP(w, req.Method, req.URL, req.Body, req.ContentLength, req.Header)
 }
+
+func (p *SynapseProxy) measure(req *http.Request) (*http.Response, error) {
+	start := time.Now()
+	resp, httpErr := p.Client.Do(req)
+
+	if p.Metrics != nil {
+		metric, err := p.Metrics.GetMetricWithLabelValues(req.URL.Path, req.Method)
```

Comment:
How does this work with the API requests that contain roomIDs or other random strings in the path?